cpp/src/DataStorm/DataElementI.cpp

```diff
@@ -18,14 +18,14 @@ namespace
 {
     DataSample toSample(const shared_ptr<Sample>& sample, const Ice::CommunicatorPtr& communicator, bool marshalKey)
     {
-        return {
-            sample->id,
-            marshalKey ? 0 : sample->key->getId(),
-            marshalKey ? sample->key->encode(communicator) : Ice::ByteSeq{},
-            chrono::time_point_cast<chrono::microseconds>(sample->timestamp).time_since_epoch().count(),
-            sample->tag ? sample->tag->getId() : 0,
-            sample->event,
-            sample->encode(communicator)};
+        return DataSample{
+            .id = sample->id,
+            .keyId = marshalKey ? 0 : sample->key->getId(),
+            .keyValue = marshalKey ? sample->key->encode(communicator) : Ice::ByteSeq{},
+            .timestamp = chrono::time_point_cast<chrono::microseconds>(sample->timestamp).time_since_epoch().count(),
+            .tag = sample->tag ? sample->tag->getId() : 0,
+            .event = sample->event,
+            .value = sample->encode(communicator)};
```

Comment:
For large structs with many fields this style, using the filed names, seems easier to read.